Update Examine custom index examples (#7611)
umbraco/UmbracoDocs: 17/umbraco-cms/reference/searching/examine/indexing.md

## Creating your own index

### A custom Umbraco content index

The following example will show how to create an index that will only include nodes based on the **Product**.

{% hint style="info" %}
We recommend that you use the existing built in ExternalIndex. You should then query based on the NodeTypeAlias instead of creating a new separate index based on that particular node type. However, should the need arise, the example below will show you how to do it.

> Take a look at our [Examine Quick Start](quick-start.md) to see some examples of how to search the ExternalIndex.

Reviewer: ```suggestion
Take a look at the [Examine Quick Start](quick-start.md) guide to see some examples of how to search the ExternalIndex.
```

The order of these registrations matters. It is important to register your index with `AddExamineLuceneIndex` before calling `ConfigureOptions`.
{% endhint %}

#### Result

![Custom product index](images/examine-management-product-index.png)

![Product document](images/examine-management-product-document.png)

### A custom index for non-Umbraco data

If you have a need, you can also use an Examine index for other data, that you aren't managing as Umbraco content.

> As an illustrative example, we can define a collection of books. Here we have a hardcoded collection. In the real-world, it's more likely these will come from a database.

Reviewer: ```suggestion
As an illustrative example, consider a collection of books. Here we have a hardcoded collection. In the real world, it's more likely these will come from a database.
```

```csharp
namespace Umbraco.Docs.Samples.Web.CustomIndexing;

public class Book
{
    public int Id { get; set; }

    public string Title { get; set; } = string.Empty;

    public string Author { get; set; } = string.Empty;

    public int PublishedYear { get; set; }
}

public static class BookData
{
    public static List<Book> GetBooks() =>
        [
            new() { Id = 1, Title = "The Great Gatsby", Author = "F. Scott Fitzgerald", PublishedYear = 1925 },
            new() { Id = 2, Title = "To Kill a Mockingbird", Author = "Harper Lee", PublishedYear = 1960 },
            new() { Id = 3, Title = "1984", Author = "George Orwell", PublishedYear = 1949 },
            new() { Id = 4, Title = "Pride and Prejudice", Author = "Jane Austen", PublishedYear = 1813 },
            new() { Id = 5, Title = "The Catcher in the Rye", Author = "J.D. Salinger", PublishedYear = 1951 }
        ];
}
```

> As with the previous example, we define an index. As this time we aren't indexing Umbraco data, we inherit directly from `LuceneIndex`:

Reviewer: ```suggestion
As with the previous example, define an index. At this time, we aren't indexing Umbraco data; we inherit directly from `LuceneIndex`:
```
